gapic/src/main/com/google/gapid/widgets/LoadingIndicator.java

```diff
@@ -186,4 +196,30 @@ public Point computeSize(int wHint, int hHint, boolean changed) {
       return new Point(SMALL_SIZE, SMALL_SIZE);
     }
   }
+
+  public class WithImage extends Widget {
```

Comment:
I think it would be a bit easier to understand and maintain if instead of extending the `Widget` class, we enhance the `Widget` version to optionally show an image on completion. Simply move the success and failure images there, in the current constructor initialize them with null, add a new constructor with the images, and in its paint method show them if they are non-null.

Reply:
Done.